Index: xfce4-vala-4.8.1/configure.ac
===================================================================
--- xfce4-vala-4.8.1.orig/configure.ac
+++ xfce4-vala-4.8.1/configure.ac
@@ -40,7 +40,7 @@ AC_ARG_WITH([default-vapidir],
 		[],
 		[with_default_vapidir=no])
 if test x"$with_default_vapidir" == xno ; then
-	AC_SUBST([vapidir], [${datarootdir}/vala-0.12/vapi])
+	AC_SUBST([vapidir], [${datarootdir}/vala-0.18/vapi])
 fi
 
 dnl Makefile outputs
Index: xfce4-vala-4.8.1/acinclude.m4
===================================================================
--- xfce4-vala-4.8.1.orig/acinclude.m4
+++ xfce4-vala-4.8.1/acinclude.m4
@@ -18,31 +18,31 @@ PKG_CHECK_EXISTS([$2 >= $3], [],
 
 dnl # M8T_CHECK_VALA(minimum_version)
 dnl #
-dnl # Check for the package libvala-0.12 and substitutes useful Vala variables.
+dnl # Check for the package libvala-0.18 and substitutes useful Vala variables.
 dnl #
 AC_DEFUN([M8T_CHECK_VALA],
 [
-PKG_CHECK_MODULES([VALA], [libvala-0.12 >= $1])
+PKG_CHECK_MODULES([VALA], [libvala-0.18 >= $1])
 M8T_VALA_PROGS()
 M8T_VAPI_DIR()
 ])
 
 dnl # M8T_VALA_PROGS()
 dnl #
-dnl # Substitutes VALAC, VAPIGEN and VALAGI from libvala-0.12 pkgconfig file.
+dnl # Substitutes VALAC, VAPIGEN and VALAGI from libvala-0.18 pkgconfig file.
 dnl #
 AC_DEFUN([M8T_VALA_PROGS],
 [
 AC_PATH_PROG([VALAC], [valac], [valac])
-AC_SUBST([VAPIGEN], `$PKG_CONFIG --variable=vapigen libvala-0.12`)
-AC_SUBST([VALAGI], `$PKG_CONFIG --variable=vala_gen_introspect libvala-0.12`)
+AC_SUBST([VAPIGEN], `$PKG_CONFIG --variable=vapigen libvala-0.18`)
+AC_SUBST([VALAGI], `$PKG_CONFIG --variable=vala_gen_introspect libvala-0.18`)
 ])
 
 dnl # M8T_VAPI_DIR()
 dnl #
-dnl # Substitutes vapidir from libvala-0.12 pkgconfig file.
+dnl # Substitutes vapidir from libvala-0.18 pkgconfig file.
 dnl #
 AC_DEFUN([M8T_VAPI_DIR],
 [
-AC_SUBST([vapidir], `$PKG_CONFIG --variable=vapidir libvala-0.12`)
+AC_SUBST([vapidir], `$PKG_CONFIG --variable=vapidir libvala-0.18`)
 ])
